### PR TITLE
Add Storage API

### LINKED ISSUE #1653: addon.storage API
Would be nice for future addons...

### PR DESCRIPTION
**Pull request by TheColaber**
**Resolves**

Resolves #1653

**Changes**

Creates API

**Reason for changes**

For the future to not use localStorage in addons anymore.

**Tests**

I tested by testing in 60FPS and scratch-messaging and both worked so yay.

### CODE REVIEW

**Comment by apple502j** [comment]
Please don't mess with storages for now, big changes coming soon

**Comment by TheColaber** [comment]
> big changes coming soon

big changes?

File: addon-api/common/Storage.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import Listenable from "./Listenable.js";
+
+/**
+ * Manages settings.
```
Comment (RedGuy12) [inline addon-api/common/Storage.js:4]: *storage

**Review by RedGuy12 — CHANGES_REQUESTED** [review]

**Comment by Norbiros** [comment]
Any progress on this? Why it is waiting? What else needs to be done?

**Comment by RedGuy12** [comment]
> Any progress on this? Why it is waiting? What else needs to be done?

@Norbiros because


> Please don't mess with storages for now, big changes coming soon

**Comment by Norbiros** [comment]
Soon > year? Okay... Can i know what changes and when they gonna be added?

**Comment by WorldLanguages** [comment]
I forgot what changes those were, but I find it unlikely we'll add an API for this before the manifest v3 migration

**Comment by TheColaber** [comment]
Closing since we'll have to deal with mv3 later anyway and this is outdated.

**Pull request closed without merging**